### PR TITLE
cast numbers to strings when accessing assertions which always yield string subjects

### LINKED ISSUE #7314: Assertions passed with a number argument should be conditionally be casted as a string on DOM string accessors
### Current behavior:

Passing a `number` to the the following assertions always fails, but casting it to a string passes.

- `have.text`
- `have.id`
- `have.data`
- `have.value`
- `have.attr`

### Desired behavior:

Numbers should automatically be casted to a string when necessary.

### Test code to reproduce

```jsx
// app.html
<div>5</div>

// app.spec.js
cy.get("div").should("have.text", 5)       // fails
cy.get("div").should("have.text", `${5}`)  // passes
```

And other combinations listed above ^^

### Versions

4.5.0

### PR DESCRIPTION
**Pull request by brian-mann**
fixes #7314 

### User facing changelog

Assertions using numbers as arguments now pass without strictly casting the number to a string:

- `have.text`
- `have.id`
- `have.data`
- `have.value`
- `have.attr`

### Additional details

```jsx
// app.html
<div>5</div>

// app.spec.js
cy.get("div").should("have.text", 5) // now passes
```

### How has the user experience changed?

Screen Shot 2020-05-12 at 4 30 34 PM

Screen Shot 2020-05-12 at 4 30 48 PM

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue been tagged with a release in ZenHub? 
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](cli/types/index.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](cli/schema/cypress.schema.json)?

### CODE REVIEW

**Comment by jennifer-shehane** [comment]
See @flotwig comment about stringifying these values. I agree there may need to be some more edge cases accounted for. https://github.com/cypress-io/cypress/issues/7314#issuecomment-627531776

**Comment by flotwig** [comment]
@jennifer-shehane I think the edge cases I outline in that comment are *impossible* to account for in an intuitive way, which is why this should probably just throw an error.

**Comment by Bkucera** [comment]
@flotwig we stringify numbers passed to `.type(...)` and don't throw an error, so I think this PR implements more consistent behavior.

File: packages/driver/test/cypress/integration/commands/assertions_spec.js [hunk 1]
```diff
@@ -1357,6 +1365,14 @@ describe('src/cy/commands/assertions', () => {
         )
       })
 
+      // https://github.com/cypress-io/cypress/issues/7314
+      it('supports a number argument', () => {
+        cy.get('.999').then(($el) => {
+          expect($el).to.have.class(999)
```
Comment (bahmutov) [inline packages/driver/test/cypress/integration/commands/assertions_spec.js:1371]: hmm, a little weird to have class as a number

**Review by kuceb — APPROVED** [review]

**Pull request merged**